lisa/tools/nvmecli.py

```diff
@@ -161,6 +161,13 @@ def get_disks(self, force_run: bool = False) -> List[str]:
 
     def get_namespace_ids(self, force_run: bool = False) -> List[Dict[str, int]]:
         nvme_devices = self.get_devices(force_run=force_run)
+        # Older versions of nvme-cli do not have the NameSpace key in the output
+        # skip the test if NameSpace key is not available
+        if not nvme_devices or "NameSpace" not in nvme_devices[0]:
+            raise SkippedException(
+                "'NameSpace' key is not available in 'nvme -list -o json' output"
```

Comment:
The message is not clear enough. It says what's verified, but not explain why the test case is skipped from the test purpose. Why is the test skipped, when the NameSpace is not available? Even the Namespace doesn't exist, it's still a nvme device.

Reply:
Older images like CentOS don't have latest nvme cli which is missing support for "Namespace" field in the output.

This is needed only for serial hot add storage testcases (with disk controller == NVMe) but not for actual NVMe testcases.
Parallel storage testcases and other tests are not impacted by this.

The "Namespace" field is used to get the LUN numbers of the remote disks which is used by all serial hot add disk tests. Without this the tests will not work on VMs with NVMe disk controller.  

Modified the exception message saying the reason and added a debug message

Reply:
Thanks for fixing, it's better, and can be improved too.

1. It needs to raise skipped exceptions in tool level, because it doesn't know how the tools will be used. Some test cases maybe skipped unexpected. Please raise general exception here, and catch/rethrow it to skipped in test cases.
2. The message should tell the details like in the review comments. Some information is important to help understanding why it's skipped like: _The "Namespace" field is used to get the LUN numbers of the remote disks which is used by all serial hot add disk tests. Without this the tests will not work on VMs with NVMe disk controller._